### PR TITLE
Feat/ps/connect front and backend

### PR DESCRIPTION
**Pull request by aliassan**
# Description

Connect the frontend with backend.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

* **Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: app/src/components/FormInputError.vue
Comment (dasarathg68) [inline app/src/components/FormInputError.vue:1]: image
@aliassan take a look at this

**Review by dasarathg68 — APPROVED** [review]

File: app/src/views/RoleCategoryView.vue [hunk 1]
```diff
@@ -5,13 +5,16 @@
       <section class="hero-container flex justify-center bg-white border rounded-xl">
         <div class="hero-content text-center bg-white p-10 w-full bg-base-200">
```
Comment (dasarathg68) [inline app/src/views/RoleCategoryView.vue:6]: Two properties set here for background
bg-white and bg-base-200

File: app/src/components/forms/roles/AddRoleCategoryForm.vue [hunk 2]
```diff
@@ -7,15 +7,29 @@
   <div class="max-h-[70vh] overflow-auto">
```
Comment (dasarathg68) [inline app/src/components/forms/roles/AddRoleCategoryForm.vue:7]:
```suggestion
  <div class="max-h-[70vh]">
```
image

**Review by hermannleboss — APPROVED** [review]
The Folder Structure is still in discussion. 
So you can merge if you feel ready.

File: app/src/components/forms/roles/AddRoleCategoryForm.vue (hunk 2 above)
Comment (aliassan) [reply app/src/components/forms/roles/AddRoleCategoryForm.vue:7]: Okay, thanks

File: app/src/views/RolesView.vue [hunk 3]
```diff
@@ -92,23 +93,45 @@ const roleCategory = ref({
...
+
+const roleCategory = ref(initRoleCategory)
 
-function navigateToRole(id: number) {
-  router.push('/roles/' + id)
+function navigateToRole(id: number | undefined) {
+  if (id) router.push('/roles/' + id)
 }
+
+const { data: _roleCategories, execute: getRoleCategoriesAPI } = useCustomFetch('role-category', {
```
[the remark this answers is not in the document]
Comment (aliassan) [reply app/src/views/RolesView.vue]: Okay, will update

File: app/src/views/RoleCategoriesView.vue [hunk 4]
```diff
@@ -65,11 +65,10 @@ const roleCategory = ref({
 })
 
 const {
-  error: roleCategoryError, 
-  data: _roleCategories, 
+  error: roleCategoryError,
+  data: _roleCategories,
```
Comment (hermannleboss) [inline app/src/views/RoleCategoriesView.vue:69]: Why this syntaxes ?

Comment (aliassan) [reply app/src/views/RoleCategoriesView.vue:69]: One is for data for displaying RoleCategoryCards with role categories from the database and the other is for creating a new role category with AddRoleCategoryForm

**Comment by dasarathg68** [comment]
@aliassan can you use the AccordionComponent.vue and then pass your accordions down as props instead?

File: app/src/components/forms/roles/AddRoleCategoryForm.vue [hunk 5]
```diff
@@ -57,7 +71,8 @@
             }
           "
         >
-          <IconMinus />
+          <!--<IconMinus />-->
```
Comment (farrellh1) [inline app/src/components/forms/roles/AddRoleCategoryForm.vue:74]:
```suggestion
```
this one not used? maybe we can delete it

File: app/src/components/forms/roles/AddRoleCategoryForm.vue [hunk 6]
```diff
@@ -68,15 +83,16 @@
...
+                    entitlementTypeId: 0,
+                    value: ''
                   }
                 ]
               })
             }
           "
         >
-          <IconPlus />
+          <!--<IconPlus />-->
```
Comment (farrellh1) [inline app/src/components/forms/roles/AddRoleCategoryForm.vue:94]:
```suggestion
```
also for this one

File: app/src/components/forms/roles/AddRoleCategoryForm.vue [hunk 7]
```diff
@@ -114,60 +130,44 @@
...
-              type: 0,
-              rule: ''
+              entitlementTypeId: 0,
+              value: ''
             })
           }
         "
       >
-        <IconPlus />
+        <!--<IconPlus />-->
```
Comment (farrellh1) [inline app/src/components/forms/roles/AddRoleCategoryForm.vue:147]:
```suggestion
```
and this and some other comments if not used

**Review by farrellh1 — COMMENTED** [review]
and this PR quite big 😅

File: app/src/components/forms/roles/AddRoleCategoryForm.vue (hunk 5 above)
Comment (aliassan) [reply app/src/components/forms/roles/AddRoleCategoryForm.vue:74]: Okay, thanks will remove those

**Pull request merged**